packages/provider_devtools_extension/lib/src/provider_screen.dart

```diff
@@ -42,71 +46,97 @@ class ProviderScreenBody extends ConsumerWidget {
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final splitAxis = Split.axisFor(context, 0.85);
+    final splitAxis = SplitPane.axisFor(context, 0.85);
 
     // A provider will automatically be selected as soon as one is detected
     final selectedProviderId = ref.watch(selectedProviderIdProvider);
     final detailsTitleText = selectedProviderId != null
         ? ref.watch(_selectedProviderNode)?.type ?? ''
         : '[No provider selected]';
 
-    ref.listen<bool>(_hasErrorProvider, (_, hasError) {
-      if (hasError) showProviderErrorBanner();
+    ref.listen<AsyncError?>(_hasErrorProvider, (_, hasError) {
+      if (hasError != null) {
+        extensionManager.postMessageToDevTools(
+            DevToolsExtensionEvent(DevToolsExtensionEventType.unknown, data: {
+          'stacktrace': hasError.stackTrace.toString(),
+          'error': hasError.toString()
+        }));
+        showProviderErrorBanner();
+      }
+    });
+    serviceManager.connectedState.addListener(() {
+      if (serviceManager.connectedState.value.connected) {
+        setServiceConnectionForProviderScreen(serviceManager.service!);
+      }
     });
```

Comment:
Listeners cannot be added inside widgets like this. You'd be causing a memory leak and accumulate more and more listeners over time.